Fix tab close button positioning

wldcordeiro opened the pull request:
Associated Issue: #2041 

### Summary of Changes

* make the tabs into inline flexboxes and remove absolute positioning from the close button.

### Test Plan

Tell us a little a bit about how you tested your patch.

Example test plan:

- [x] open debugger
- [x] open some tabs
- [x] see tab close button is in proper position for `ltr` mode (rtl will be in separate PR as there is more broken there.)

### Screenshots/Videos

#### Before
![screenshot_20170216_232152](https://cloud.githubusercontent.com/assets/580982/23060568/7fcc0f34-f4bb-11e6-9e78-82ca30cd7389.png)

#### After
![screenshot_20170217_030222](https://cloud.githubusercontent.com/assets/580982/23061047/85590c2a-f4bd-11e6-9375-b0c618b4257a.png)

Seems like the close button is a little too close to the tab title. 🤔

clarkbw commented:
Yeah, a little spacing from the title would be good.  Awesome find!

jasonLaster commented:
Thanks! How's this look with RTL?

princiya commented:
So the bug is coming from this (potential) explanation: https://github.com/devtools-html/debugger.html/pull/1976/files/368101b4c46ec34a6a2ba129fa7969d6ed65b4be#diff-d9abd0398fe2e0fa5793ba7d100231f5L50
and here: https://github.com/devtools-html/debugger.html/pull/1976/files/368101b4c46ec34a6a2ba129fa7969d6ed65b4be#diff-d9abd0398fe2e0fa5793ba7d100231f5L110

Currently we have,
`.source-tab .close-btn {
  position: absolute;
  top: 3px;
  offset-inline-end: 4px;
  visibility: hidden;
}`

and writing this explicitly as below fixes the issue.
`html[dir="ltr"] .source-tab .close-btn {
  offset-inline-end: 4px;
}`

`html[dir="rtl"] .source-tab .close-btn {
  offset-inline-start: 4px;
}`

jasonLaster commented:
Thanks @princiya.

I looked at it for a bit and I think we'll want to touch up the margin/padding for the close button and source tab.

We should also look at the pretty print button:
* it's also position absolute
* we shouldn't reserve the 20px space unless it's a pretty source
* we should vertically align the icon so it lines up w/ the text buttons

jasonLaster commented:
Here's a good screenshot of the two problems I see:

1. the close button should have some more padding
2. the pretty print button should be better aligned

![screen shot 2017-02-21 at 11 15 26 am](https://cloud.githubusercontent.com/assets/254562/23173446/2b00f27a-f827-11e6-9551-6d5bae4669f3.png)

wldcordeiro commented:
Fixes were just pushed I added some padding to the close button and fixed up the alignment of the pretty print icon.

![screenshot_20170222_034202](https://cloud.githubusercontent.com/assets/580982/23208111/03c76128-f8b1-11e6-98c5-27d697a8ebc7.png)
![screenshot_20170222_034223](https://cloud.githubusercontent.com/assets/580982/23208112/03c854ac-f8b1-11e6-86b5-0b367e9de690.png)

clarkbw commented on src/components/Editor/Tabs.css:
```diff
@@ -94,13 +90,20 @@ html[dir="rtl"] .source-tab.active {
+}
+
+html[dir="ltr"] .source-tab .close-btn {
+  margin-left: 6px;
```
I don't know this stuff well, but could this be `margin-inline-start` instead?

jasonLaster replied:
I think so. I'm going to pull in the latest bidirection and then test

jasonLaster commented:
This needs the launchpad bump to look right
